graph/cycle_check_undirected_graph.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file cycle_check_undirected graph.cpp
+ *
+ * @brief  Disjoint Set (Or Union-Find) algorithm to check for cycle in a undirected graph.
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible) and proper documentation.
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

Reply:
I have added the Wikipedia link .